### PR TITLE
Markdown Treesitter Groups

### PR DESCRIPTION
**Pull request by evantravers**
Finishes #63

### CODE REVIEW

**Comment by evantravers** [comment]
image

image

**Comment by mcchrish** [comment]
Any idea why there are some underlined words in the code blocks? It seems the spellchecker is activated in those areas.
Screen Shot 2021-12-17 at 3 52 32 PM

**Review by mcchrish — CHANGES_REQUESTED** [review]
Markdown is in a unique position that it's not really close to any typical programming language. That's why I think these linkings should be applied specifically to tree-sitter markdown hl groups. e.g. it should be `markdownTSLiteral` instead of the general `TSLiteral`.

**Comment by mcchrish** [comment]
I actually like how the non-treesitter handles links where the link text is underlined while the URI is highlighted withly less contrast i.e. linked to `Number`.

With TS:
Screen Shot 2021-12-17 at 4 05 30 PM

Without TS:
Screen Shot 2021-12-17 at 4 05 01 PM

Is it possible to do this in TS? According to TS playground, there is no highlight group applied to the link text. We might need to patch this [upstream](https://github.com/nvim-treesitter/nvim-treesitter).

**Comment by evantravers** [comment]
I agree whole heartedly, and am excited that your change to tree-sitter was approved so fast.

**Comment by evantravers** [comment]
I have not seen reference to these markdown treesitter groups like `markdownTSLiteral`… are you telling me to change my PR to something that exists or are you wishing something existed so that I could change my PR?

**Comment by mcchrish** [comment]
Tree-sitter creates specific versions of the groups per language and are linked to the general group. e.g. `vimTSLiteral`, `javascriptTSLiteral`, `markdownTSLiteral` and so on. This quite useful when we want to only target specific highlights of a language. You can see these groups when running `:highlights`.

My request is that we target these markdown specific groups instead. We have to create new groups for this. Example:

```lua
markdownTSLiteral { Delimiter };
```

**Comment by evantravers** [comment]
Thanks for the explanation… I was confused that there was literally _no_ mention of `markdownTSLiteral` anywhere on github.

I'll make those changes and ping you again later this weekend.

**Comment by evantravers** [comment]
Ok… pulled out the original group additions/changes, made a markdown specific group, added to both files.

I don't love the revert (looks ugly in git history)… so when we approve the changes I'll squash and rebase before merge. Thoughts?

**Comment by mcchrish** [comment]
This looks great! One last thing I want to request is the link styling I mentioned here: https://github.com/mcchrish/zenbones.nvim/pull/64#issuecomment-996518952

File: lua/zenbones/specs/dark.lua [hunk 1]
```diff
@@ -253,6 +253,17 @@ local function generate(p, opt)
 			TSWarning               { WarningMsg },
 			TSDanger                { Error },
 
+
```
Comment (mcchrish) [inline lua/zenbones/specs/dark.lua:256]: Single space so it will look consistent.

Comment (evantravers) [reply lua/zenbones/specs/dark.lua:256]: 👍

**Comment by evantravers** [comment]
image
image

After changes.

**Review by mcchrish — APPROVED** [review]

**Comment by evantravers** [comment]
`git reset --soft` for a cleaned up commit. Same changes.

**Comment by mcchrish** [comment]
Thanks for the contribution! 🚀

**Pull request merged**